sympy/core/relational.py

```diff
@@ -302,17 +302,37 @@ def __new__(cls, lhs, rhs=0, **options):
             # If expressions have the same structure, they must be equal.
             if lhs == rhs:
                 return S.true
-
-            # If appropriate, check if the difference evaluates.  Detect
-            # incompatibility such as lhs real and rhs not real.
-            if lhs.is_complex and rhs.is_complex:
-                r = (lhs - rhs).is_zero
-                if r is not None:
-                    return _sympify(r)
-
-            # If expression have both Boolean terms
-            if all(isinstance(i, BooleanAtom) for i in (rhs, lhs)):
-                return S.false  # equal args already evaluated
+            elif all(isinstance(i, BooleanAtom) for i in (rhs, lhs)):
+                return S.false
+
+            # check finiteness
+            fin = L, R = [i.is_finite for i in (lhs, rhs)]
+            if None not in fin:
+                if L != R:
+                    return S.false
+                if L is False:
+                    return S.true
+
+            # see if the difference evaluates
+            if all(isinstance(i, Expr) for i in (lhs, rhs)):
+                dif = lhs - rhs
+                z = dif.is_zero
+                if z is not None:
+                    if z is False and dif.is_commutative:  # issue 10728
+                        return S.false
+                    if z:
+                        return S.true
```

Comment:
```
if any(isinstance(lhs, Pow) for (lhs, rhs)):
     pass
```
that's valid because normal Pow calculated in before condition.for example:
```
Eq(2**2, 4)
```
we can put this condition with below condition also.